src/modules/Channel/components/Message/hooks/useIsElementInViewport.ts

```diff
@@ -0,0 +1,20 @@
+import React, { useEffect, useState } from 'react';
+
+export const useIsElementInViewport = (elementRef: React.MutableRefObject<any>) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const callback = (entries: IntersectionObserverEntry[]) => {
+    const [entry] = entries;
+    setIsVisible(entry.isIntersecting);
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(callback);
+    if (elementRef.current) {
+      observer.observe(elementRef.current);
+    }
+    return () => observer.disconnect();
+  }, [elementRef]);
+
+  return { elementRef, isVisible };
```

Comment:
Guess we don't need to re-return `elementRef` since there's no places we use it.